apps/sms/test/marionette/notification_test.js

```diff
@@ -225,7 +226,63 @@ marionette('Message notification tests', function() {
 
       openNotification();
 
-      assertMessagesIsInCorrectState();
+      assertMessagesIsInCorrectStateFromMessage(smsMessage);
+    });
+  });
+
+  suite('Display a conversation from a notification', function() {
+    var messagesStorage;
+    var thread1, thread2;
+
+    setup(function() {
+      thread1 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['999']
+      });
+      thread2 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['888']
+      });
+
+      messagesStorage = [ thread1, thread2 ];
+    });
+
+    test('Clicking a notification while a conversation is loading', function() {
+      messagesApp.launch();
+      storage.setMessagesStorage(messagesStorage);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+
+      var inbox = new InboxView(client);
+      var conversation = inbox.goToConversation(
+        thread1.id, { shouldWait: false }
+      );
+
+      // Receive 'notification' system message and fill up the storage.
+      messagesApp.sendSystemMessage('notification', {
+        clicked: true,
+        data: { threadId: thread2.id, id: thread2.messages[0].id }
+      });
+
+      conversation.accessors.waitToAppear();
+      var thread2Ids = thread2.messages.map(
+        function(message) { return message.id; }
+      );
+
+      // Our testing strategy is:
+      // 1. Wait that all messages are rendered for thread2.
+      thread2Ids.forEach(function(messageId) {
```

Comment:
Ah yeah, it waits for the default timeout and fails if it can't find, likely was thinking about ```findElements```, sorry, please ignore ```Looks like findMessage doesn't guarantee that message is rendered already so it can return null```, but I can't recall why I added null check there :/